pkg/api/graph/types.go

```diff
@@ -47,6 +51,72 @@ const (
 	ReferencedImageLayerGraphEdgeKind
 )
 
+var kindToGraphKind = map[reflect.Type]interface{}{}
+
+func init() {
+	kindToGraphKind[reflect.TypeOf(&image.ImageStream{})] = ImageStreamGraphKind
+	kindToGraphKind[reflect.TypeOf(&image.Image{})] = ImageGraphKind
+	kindToGraphKind[reflect.TypeOf(&build.BuildConfig{})] = BuildConfigGraphKind
+	kindToGraphKind[reflect.TypeOf(&build.Build{})] = BuildGraphKind
+	kindToGraphKind[reflect.TypeOf(&deploy.DeploymentConfig{})] = DeploymentConfigGraphKind
+	kindToGraphKind[reflect.TypeOf(&kapi.Service{})] = ServiceGraphKind
+	kindToGraphKind[reflect.TypeOf(&kapi.Pod{})] = PodGraphKind
+	kindToGraphKind[reflect.TypeOf(&kapi.ReplicationController{})] = ReplicationControllerGraphKind
+}
+
+func GetUniqueNamespaceNodeName(obj runtime.Object) UniqueName {
+	return getUniqueNodeName(obj, true)
+}
+
+func GetUniqueRootScopedNodeName(obj runtime.Object) UniqueName {
+	return getUniqueNodeName(obj, false)
+}
+
+func getUniqueNodeName(obj runtime.Object, namespaced bool) UniqueName {
+	objType := reflect.TypeOf(obj)
```

Comment:
Honestly, that's a pretty weak argument.  Test cases can do that too.

> On Jun 22, 2015, at 12:48 PM, David Eads <notifications@github.com> wrote:
> 
> In pkg/api/graph/types.go:
> 
> > +	kindToGraphKind[reflect.TypeOf(&deploy.DeploymentConfig{})] = DeploymentConfigGraphKind
> > +	kindToGraphKind[reflect.TypeOf(&kapi.Service{})] = ServiceGraphKind
> > +	kindToGraphKind[reflect.TypeOf(&kapi.Pod{})] = PodGraphKind
> > +	kindToGraphKind[reflect.TypeOf(&kapi.ReplicationController{})] = ReplicationControllerGraphKind
> > +}
> > +
> > +func GetUniqueNamespaceNodeName(obj runtime.Object) UniqueName {
> > +	return getUniqueNodeName(obj, true)
> > +}
> > +
> > +func GetUniqueRootScopedNodeName(obj runtime.Object) UniqueName {
> > +	return getUniqueNodeName(obj, false)
> > +}
> > +
> > +func getUniqueNodeName(obj runtime.Object, namespaced bool) UniqueName {
> > +	objType := reflect.TypeOf(obj)
> How is this better than a switch?
> 
> A map allows for future inspection of registered keys so that we can figure out how many api resources cannot be referenced in the map.
> 
> —
> Reply to this email directly or view it on GitHub.
>